x-pack/solutions/security/plugins/security_solution/public/sourcerer/containers/index.tsx

```diff
@@ -67,10 +66,10 @@ export const useSourcererDataView = (
     if (selectedDataView == null || missingPatterns.length > 0) {
       // old way of fetching indices, legacy timeline
       setLegacyPatterns(selectedPatterns);
-    } else {
+    } else if (legacyPatterns.length > 0) {
```

Comment:
A new `[]` was being created on every run of this hook, thereby updating state, thereby re-rendering every component where this hook is being called